### PR TITLE
Add WYSIWYG MD-editor #143

### LINKED ISSUE #143: Use a simple WYSIWYG editor for group / ticket description
Preferred is markdown without embedded HTML.

### PR DESCRIPTION
**Pull request by triszt4n**
Closes #143 

### What's new:
* The textareas in group/new and ticket/new are based upon simpleMDE markdown editors.
* The backend stores description in markdown.
* Descriptions are converted and sanitized before sending them in response.
* Groups main page gets the descriptions abbreviated.

### CODE REVIEW

File: src/components/groups/group.service.ts [hunk 1]
```diff
@@ -1,13 +1,32 @@
...
+  const cleanHtml = sanitizeHtml(dirtyHtml, {
+    transformTags: {
+      'h1': 'h3',
+      'h2': 'h4',
+      'h3': 'h5',
+      'h4': 'h5'
+    }
+  })
+  return cleanHtml
+}
```
Comment (OmTheTurtle) [inline src/components/groups/group.service.ts:18]: This code block is duplicated, you can extract it to a separate file and put it under `src/util`.

File: src/components/tickets/ticket.service.ts [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+    }
+  })
+  return cleanHtml
+}
 
 export const getTickets = async (req: Request, _res: Response, next: NextFunction) => {
   req.tickets = await Ticket.query().orderBy('createdAt', 'ASC')
+  req.tickets.forEach((ticket) => {
+    ticket.description = formatMdToSafeHTML(ticket.description)
+  })
```
Comment (OmTheTurtle) [inline src/components/tickets/ticket.service.ts:23]: This I think can be written like this:
```
req.tickets = (await Ticket.query().orderBy('createdAt', 'ASC')).map(ticket => {
  ticket.description = formatMdToSafeHTML(ticket.description)
  return ticket
}
```
I think using a `map` may be a little more descriptive in this case.

File: src/components/groups/group.service.ts [hunk 3]
```diff
@@ -1,13 +1,32 @@
...
 export const getGroups = async (req: Request, res: Response, next: NextFunction) => {
   const page = parseInt(req.query.page ?? 0)
   const limit = 10
   const pageObject = await Group.query().orderBy('createdAt', 'DESC').page(page, limit)
   req.groups = pageObject.results
+
+  req.groups.forEach((group) => {
+    group.description = group.description.slice(0, 50) + ' ...'
+    group.description = formatMdToSafeHTML(group.description)
```
Comment (OmTheTurtle) [inline src/components/groups/group.service.ts:27]: You can combine these two calls into one, and refactor the `forEach` to a map as in `ticket.service.ts`.

File: src/components/tickets/ticket.service.ts (hunk 2 above)
Comment (triszt4n) [reply src/components/tickets/ticket.service.ts:23]: Completely forgot about .map. Thanks!

File: public/js/md-editor.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    className: 'chars',
+    defaultValue: (el) => {
+      el.innerHTML = '0 / ' + charLimit
+    },
+    onUpdate: (el) => {
+      characterCount = easyMDE.value().length
+      listenLength(el, characterCount)
+    }
+  }]
+})
```
[the remark this answers is not in the document]
Comment (triszt4n) [reply public/js/md-editor.js:38]: Gonna mend them it in next commit!

[the remark this answers is not in the document]
Comment (triszt4n) [reply public/js/md-editor.js:38]: The other thing: there's an issue with webpack integration. I guess, most of the cdn-s will be removed from project when webpack integration is complete. But I'mma quote your comment onto the issue (#107)

[the remark this answers is not in the document]
Comment (falusi94) [reply public/js/md-editor.js:38]: Ah, great, then I said nothing!

**Comment by triszt4n** [comment]
All the UI modifications are my suggestions. We don't have to accept these. Just made some modifications, to make the UI look better or look better on mobile actually (e.g.: main page room buttons)

File: src/components/groups/group.service.ts [hunk 5]
```diff
@@ -21,6 +25,7 @@ export const getGroup = async (req: Request, res: Response, next: NextFunction)
     .withGraphFetched('users')
 
   if (group) {
+    group.description = formatMdToSafeHTML(group.description)
     req.group = group
```
Comment (OmTheTurtle) [inline src/components/groups/group.service.ts:29]: Just another little thing you can do in these situations is using the object rest operator:
```
req.group = { ...group, description: formatMdToSafeHTML(group.description) }
```

It's perfectly fine as it is, it's just another way to do it.

Comment (triszt4n) [reply src/components/groups/group.service.ts:29]: Spicy, gonna use it! Thanks!

**Pull request merged**